src/Shared/FileMatcher.cs

```diff
@@ -180,18 +180,33 @@ internal static bool HasWildcards(string filespec)
         }
 
         /// <summary>
-        /// Determines whether the given path has any wild card characters or any semicolons.
+        /// Determines whether the given path has any wild card characters or semicolons.
+        /// </summary>
+        internal static bool HasWildcardsOrSemicolon(string filespec)
+        {
+            return -1 != filespec.LastIndexOfAny(s_wildcardAndSemicolonCharacters);
```

Comment:
Is there any significant perf difference between using `Aggregate` and `-1 != filespec.LastIndexOfAny(s_wildcardAndSemicolonCharacters)`?

Reply:
Were you thinking Aggregate with a function like Aggregate(false, (acc, ch) => acc || s_wildcardAndSemicolonCharacters.Contains(ch))? That would almost certainly be slower than this.

Reply:
I was thinking the difference between `s_propertyReferences.Aggregate(false, (current, propertyReference) => current | filespec.Contains(propertyReference));` and `-1 != filespec.LastIndexOfAny(s_propertyReferences);`

If the former is more efficient, we can change `HasWildcardsOrSemicolon` to do the same.

Reply:
I believe LastIndexOf is going to be faster but I had to use the aggregate for the other because the s_wildcardAndSemicolonCharacters are `char[]` and the s_propertyReferences are `string[]`. Would need to setup a microbenchmark to validate.

Reply:
Ah! I didn't realize that we couldn't replicate what was already there. This looks fine to me 👍

Reply:
Forgind improved it further now to use Any() :)